src/Features/Core/Portable/Workspace/ProjectCacheService.cs

```diff
@@ -150,12 +157,12 @@ private void DisableCaching(ProjectId key, Cache cache)
             }
         }
 
-        private class Cache : IDisposable
+        private sealed class Cache : IDisposable
         {
             internal int Count;
             private readonly ProjectCacheService _cacheService;
             private readonly ProjectId _key;
-            private ConditionalWeakTable<object, object> _cache = new ConditionalWeakTable<object, object>();
+            private ConditionalWeakTable<object, object?>? _cache = new ConditionalWeakTable<object, object?>();
```

Comment:
💡 This should not be nullable. The assignment of `_cache` to null should be moved from `FreeOwnerEntries` to `Dispose` and a suppression added there.

Reply:
Why? We check `_cache` for non-null before it's used and throw `ObjectDisposeException`.